test/mocha/mock.plugin.js

```diff
@@ -0,0 +1,96 @@
+/*!
+ * Copyright (c) 2017-2018 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+module.exports = {
+  type: 'service',
+  api: {
+    index: async ({createIndexes, collections}) => {
+      should.exist(createIndexes);
+      createIndexes.should.be.a('function');
+      should.exist(collections);
+      collections.should.an('object');
+      should.exist(collections.blockCollection);
+      collections.blockCollection.should.be.a('string');
+      should.exist(collections.eventCollection);
+      collections.eventCollection.should.be.a('string');
+      should.exist(collections.operationCollection);
+      collections.operationCollection.should.be.a('string');
+      return createIndexes([{
+        collection: collections.operationCollection,
+        fields: {'operation.record.type': 1},
+        options: {unique: false, background: false, name: 'fooIndex'}
+      }]);
+    },
+    storage: {
+      operations: {
+        // NOTE: do not use arrow functions here because this function is to be
+        // bound to the class instance
+        query: function({maxBlockHeight, query}, callback) {
```

Comment:
Since this is new stuff, I'd prefer we use promises so we don't have to deal with two different APIs in the future.

Reply:
I'm going to update the mock plugin to use async/await.  However, the plugin API does not enforce any particular style or method names.  We've already implement an async plugin for a POC.  

I think the common case is that an agent plugin will be coupled with a storage plugin.  The agent plugin knows about and utilizes the custom methods added by the storage plugin.